From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-compaction-update-the-cc-nr_migratepages-when-allocating-or-freeing-the-freepages-fix
Date: Fri Feb  9 03:51:53 PM PST 2024

fix it for the dropping of "mm/compaction: optimize >0 order folio
compaction with free page split"

Cc: Baolin Wang <baolin.wang@linux.alibaba.com>
Cc: Mel Gorman <mgorman@techsingularity.net>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/compaction.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/compaction.c~mm-compaction-update-the-cc-nr_migratepages-when-allocating-or-freeing-the-freepages-fix
+++ a/mm/compaction.c
@@ -1798,7 +1798,7 @@ static struct folio *compaction_alloc(st
 	dst = list_entry(cc->freepages.next, struct folio, lru);
 	list_del(&dst->lru);
 	cc->nr_freepages--;
-	cc->nr_migratepages -= 1 << order;
+	cc->nr_migratepages -= 1 << folio_order(src);
 
 	return dst;
 }
@@ -1814,7 +1814,7 @@ static void compaction_free(struct folio
 
 	list_add(&dst->lru, &cc->freepages);
 	cc->nr_freepages++;
-	cc->nr_migratepages += 1 << order;
+	cc->nr_migratepages += 1 << folio_order(dst);
 }
 
 /* possible outcome of isolate_migratepages */
_
